Simple list widget (#1496)
kyma-project/busola: docs/extensibility/form-widgets.md

#### Example

```json
[
  {
    "path": "spec.service",
    "widget": "FormGroup"
  },
  {
    "path": "spec.service.host"
  },
  {
    "path": "spec.service.port"
  }
]
```

<img src="./assets/form-widgets/FormGroup.png" alt="Example of a FormGroup widget" style="border: 1px solid #D2D5D9">

### GenericList

GenericList widgets render an `array` as a list of collapsible sections with their own sub-forms. An **add** button is present to add new entries.

#### Example

```json
[
  {
    "path": "spec.services",
    "widget": "GenericList"
  },
  {
    "path": "spec.services[].host"
  },
  {
    "path": "spec.services[].port"
  }
]
```

<img src="./assets/form-widgets/GenericList.png" alt="Example of a GenericList widget" style="border: 1px solid #D2D5D9">

### SimpleList

> SimpleList widgets render an `array` as a table with rows representing data items and columns representing different fields. New items are added automatically when new entries are typed it.

Reviewer: ```suggestion
SimpleList widgets render an `array` as a table with rows representing data items and columns representing different fields. New items are added automatically when new entries are typed in.
```
Typo here?

> This type of field is only suitable for simple data types and can contain more complex structures in it's items.

Reviewer: ```suggestion
This type of field is only suitable for simple data types and can contain more complex structures in its items.
```

Author: of course

#### Example

```json
[
  {
    "path": "spec.services",
    "widget": "SimpleList"
  },
  {
    "path": "spec.services[].host"
  },
  {
    "path": "spec.services[].port"
  }
]
```

<img src="./assets/form-widgets/SimpleList.png" alt="Example of a SimpleList widget" style="border: 1px solid #D2D5D9">

#### Scalar values

> When array items are scalar instead of an object, no header with a field title will be rendered in the resulting table.

Reviewer: ```suggestion
When an array item is a scalar instead of an object, no header with a field title will be rendered in the resulting table.
```
Or `when array items are scalars instead of objects`. Unless you mean that they're either individual scalars or one collective object...?

Author: it's an array, whose items are scalar

Reviewer: I have one more question. I'll sync with you.

Reviewer: Clarified :) I edited my suggestion.
